### PR TITLE
Fix ROM Struct examples

### LINKED ISSUE #685: Broken example code, unexpected auto_struct behavior
## Describe the bug

I made a script of the example code, and it seems that the example code for the auto-generated structs no longer reflects reality, and as a result I am very confused about how to setup my custom name-spaced structs, when not even the auto-generated ones work as expected.

The example code I mostly copied is here: https://api.rom-rb.org/rom/ROM/Struct
Note that as written the example code doesn't work _at all_, as it was missing the `conf.relation(:users)`, so I added that.  Even with that, the example code still doesn't work at all, and raises the error:
`undefined method 'schema' for ROM::OpenStruct:Class (NoMethodError)`

## To Reproduce

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'
  gem 'rom'
  gem 'rom-sql'
  gem 'sqlite3'
end

rom = ROM.container(:sql, 'sqlite::memory') do |conf|
  conf.default.create_table(:users) do
    primary_key :id
    column :name, String
  end

  conf.relation(:users) do
  end
end

class UserRepo < ROM::Repository[:users]
end

user_repo = UserRepo.new(rom)

# get auto-generated User struct
model = user_repo.users.mapper.model

puts "model: #{model}"
# EXPECTED
# model: ROM::Struct::User
#
# ACTUAL
# model: ROM::OpenStruct

puts model.schema.key(:id)
# EXPECTED:
# #<Dry::Types[id: Nominal<Integer meta={primary_key: true, source: :users}>]>
#
# ACTUAL:
# undefined method `schema' for ROM::OpenStruct:Class (NoMethodError)
```

## My environment

- Affects my production application: **NOT YET** (Implementing ROM as part of a new service)
- Ruby version: 2.7.8 through 3.2.2
- OS: macOS Ventura 13.4

### PR DESCRIPTION
**Pull request by cllns**
Fixes #685

Also adds a **NOTE** in front of the caveat for ROM structs which was also discussed in #610

### CODE REVIEW

File: core/lib/rom/struct.rb [hunk 1]
```diff
@@ -33,6 +34,8 @@ module ROM
   #   end
   #
   #   class UserRepo < ROM::Repository[:users]
+  #     schema(infer: true)
+  #     auto_struct true
```
Comment (flash-gordon) [inline core/lib/rom/struct.rb:38]: these are settings for relations, not repos

File: core/lib/rom/struct.rb [hunk 2]
```diff
@@ -43,11 +46,11 @@ module ROM
   #
   #   # see struct's schema attributes
   #
-  #   # model.schema.key(:id)
-  #   # => #<Dry::Types[id: Nominal<Integer meta={primary_key: true, source: :users}>]>
+  #   model.schema.key(:id)
+  #   # => #<Dry::Types[id: Nominal<Integer meta={primary_key: true, alias: nil, source: :users}>]>
```
Comment (flash-gordon) [inline core/lib/rom/struct.rb:50]: I think omitting insignificant meta keys is fine here

File: core/lib/rom/struct.rb (hunk 1 above)
Comment (cllns) [reply core/lib/rom/struct.rb]: Ah, apologies. I made sure to test the whole script but put it in the wrong place when adding it to the comment. Fixed now

File: core/lib/rom/struct.rb (hunk 2 above)
Comment (cllns) [reply core/lib/rom/struct.rb]: Gotcha, just removed `alias`

**Review by flash-gordon — APPROVED** [review]

**Comment by flash-gordon** [comment]
thank you!

**Comment by Ahmadtaha1122** [comment]
👍

**Pull request merged**